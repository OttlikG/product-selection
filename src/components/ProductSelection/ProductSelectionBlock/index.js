
import React from 'react'

import Sports from './Sports'
import News from './News'
import Basket from './Basket'

// import classes from './index.scss'

export default function ProductSelectionBlock (props) {
  const { sports, news, selectedProducts, selectProductById, push } = props;

  return (
    <div>
      <div className="row">
        <div className="col-xs-4"><Sports data={sports} selectProductById={selectProductById} /></div>
        <div className="col-xs-4"><News data={news} selectProductById={selectProductById} /></div>
        <div className="col-xs-4"><Basket data={selectedProducts} selectProductById={selectProductById} push={push} /></div>
      </div>
    </div>
  )
}
