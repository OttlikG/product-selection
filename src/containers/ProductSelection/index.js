
import React, { Component }     from 'react'
import { connect }              from 'react-redux';
import { push }                 from 'react-router-redux'
import ProductSelectionBlock    from 'components/ProductSelection/ProductSelectionBlock'
import productionActions        from 'redux/actions/products'
import authActions              from 'redux/actions/auth'
import { newsSelector, sportsSelector, selectedProducts } from 'redux/reducers/products'
// import classes                  from './index.scss'

class ProductSelection extends Component {

  componentDidMount () {
    const { readToken } = this.props;
    readToken()
  }

  componentWillReceiveProps (nextProps) {
    const { getProducts, productionSelection, customerId } = nextProps;
    const isCustomerIdChanged = customerId !== this.props.customerId

    // if (isCustomerIdChanged) getProducts()
    if (isCustomerIdChanged) productionSelection()
  }

  render () {
    return (
      <div>
        <ProductSelectionBlock {...this.props} />
      </div>
    )
  }
}

export function mapStateToProps ({ products, auth }) {
  return {
    sports: sportsSelector(products),
    news: newsSelector(products),
    selectedProducts: selectedProducts(products),
    customerId: auth.customerId
  }
}

const actions = {
  ...productionActions,
  ...authActions,
  push,
}

function mergeProps(stateProps, dispatchProps, ownProps) {
  const { customerId } = stateProps
  const { productionSelection } = dispatchProps

  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    productionSelection: () => productionSelection(customerId)
  }
}

export default connect(mapStateToProps, actions, mergeProps)(ProductSelection)
