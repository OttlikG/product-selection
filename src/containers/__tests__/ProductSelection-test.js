
import React from 'react'
import { ProductSelection } from '../ProductSelection'

jest.mock('components/ProductSelection/ProductSelectionBlock', () => 'ProductSelectionBlock')

describe('ProductSelection', () => {
  let props, component

  beforeEach(() => {
    props = {
      customerId: 'aaa',
      readToken: jest.fn(),
      productionSelection: jest.fn(),
    }

    component = render(<ProductSelection {...props} />)
  })

  describe('componentDidMount', () => {
    it('should call readToken', () => {
      expect(props.readToken).toHaveBeenCalledTimes(1)
    })
  })

  describe('componentWillReceiveProps', () => {
    it('should call productionSelection if cutomerId changed', () => {
      const instance = component.getInstance()
      props.customerId = 'bbb'

      instance.componentWillReceiveProps(props)
      expect(props.productionSelection).toHaveBeenCalledTimes(1)
    })
  })
})
