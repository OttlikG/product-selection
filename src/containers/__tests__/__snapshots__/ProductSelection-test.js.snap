// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSelection componentDidMount should call readToken 1`] = `
<div>
  <ProductSelectionBlock
    customerId="aaa"
    productionSelection={[Function]}
    readToken={[Function]}
  />
</div>
`;

exports[`ProductSelection componentWillReceiveProps should call productionSelection if cutomerId changed 1`] = `
<div>
  <ProductSelectionBlock
    customerId="aaa"
    productionSelection={[Function]}
    readToken={[Function]}
  />
</div>
`;
